### PR TITLE
adding keyboard: gothic70

### PR DESCRIPTION
**Pull request by worldspawn00**
## Description

adding keyboard: gothic70
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/wsk/gothic70/config.h [hunk 1]
```diff
@@ -0,0 +1,69 @@
+#ifndef CONFIG_H
+#define CONFIG_H
```
Comment (Duckle29) [inline keyboards/wsk/gothic70/config.h]: Please use `#pragma once` instead of ifndef guards :)

File: keyboards/wsk/gothic70/config.h [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+#define LOCKING_RESYNC_ENABLE
+
+/* If defined, GRAVE_ESC will always act as ESC when CTRL is held.
+ * This is userful for the Windows task manager shortcut (ctrl+shift+esc).
+ */
+
+ #define GRAVE_ESC_CTRL_OVERRIDE
+
+#endif
```
Comment (Duckle29) [inline keyboards/wsk/gothic70/config.h]:
```suggestion
```

File: keyboards/wsk/gothic70/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,83 @@
+
+#include QMK_KEYBOARD_H
+
+
```
Comment (Duckle29) [inline keyboards/wsk/gothic70/keymaps/default/keymap.c]:
```suggestion
// Defines names for use in layer keycodes and the keymap
enum layer_names { 
    _BASE,
    _FN, 
    _MEDIA };
```

Using an enum for layernames can make the keyboards more readable :)

File: keyboards/wsk/gothic70/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+// Tap Dance Declarations
+enum {
+    TD_SCAPS = 0
+};
+
+#define SCAPS_LG TD(TD_SCAPS)
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+  [0] = LAYOUT(
```
Comment (Duckle29) [inline keyboards/wsk/gothic70/keymaps/default/keymap.c]:
```suggestion
  [_BASE] = LAYOUT(
```

File: keyboards/wsk/gothic70/keymaps/default/keymap.c [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+    KC_ESC,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_GRV,  KC_BSPC,          KC_PSCR, \
+    KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,             KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,          KC_INS, \
+    KC_LCTL, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,             KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT, KC_ENT,                    KC_DEL,  \
+    SCAPS_LG, KC_Z,    KC_X,    KC_C,   KC_V,    KC_B,    MO(2),  KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT,          KC_UP,                     \
+    KC_LCTL, KC_LGUI,  KC_LALT,         KC_SPC,  MO(1),                        KC_BSPC,           KC_RALT, KC_RCTL,          KC_LEFT, KC_DOWN, KC_RGHT           \
+  ),
+
+  [1] = LAYOUT(
```
Comment (Duckle29) [inline keyboards/wsk/gothic70/keymaps/default/keymap.c]:
```suggestion
  [_FN] = LAYOUT(
```

File: keyboards/wsk/gothic70/keymaps/default/keymap.c [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+    KC_GRV, KC_F1, KC_F2, KC_F3, KC_F4, KC_F5, KC_F6, KC_F7, KC_F8, KC_F9, KC_F10, KC_F11, KC_F12, _______, _______,                        KC_PAUS, \
+    _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,           _______, \
+    _______, RGB_MOD,  RGB_HUI, RGB_SAI,  RGB_VAI, _______, _______, _______, _______, _______, _______, _______, _______,                    _______, \
+    _______, RGB_TOG, RGB_HUD, RGB_SAD,  RGB_VAD, _______, _______, _______, _______, _______, _______, _______,  _______,          KC_PGUP,                   \
+    _______, _______, _______,          _______, _______,          _______,          _______,          _______,           KC_HOME, KC_PGDN, KC_END           \
+  ),
+
+  [2] = LAYOUT(
```
Comment (Duckle29) [inline keyboards/wsk/gothic70/keymaps/default/keymap.c]:
```suggestion
  [_MEDIA] = LAYOUT(
```

File: keyboards/wsk/gothic70/keymaps/default/keymap.c [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+    _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, RESET, \
+    _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,           _______,  \
+    _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, KC_MPLY,                   _______, \
+    _______, _______, _______, _______, _______, _______,   _______, _______, _______, _______, _______, _______, _______,          KC_VOLU,                   \
+    _______, _______, _______,          _______, _______,          _______,          _______,          _______, KC_MPRV, KC_VOLD, KC_MNXT           \
+  )
+};
+
+void matrix_init_user(void) {
+  // set CapsLock LED to output and lowprimekb
```
Comment (Duckle29) [inline keyboards/wsk/gothic70/keymaps/default/keymap.c]:
```suggestion
  // set CapsLock LED to output and off (active low)
```
Not sure the primekb was meant to be there. Also, you're explicitly setting the pins high, so I assume the "off" would be more accurate?

File: keyboards/wsk/gothic70/keymaps/default/keymap.c [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+  setPinOutput(F5);
+  writePinHigh(F5);
+  // set NumLock LED to output and low
```
Comment (Duckle29) [inline keyboards/wsk/gothic70/keymaps/default/keymap.c]:
```suggestion
  // set NumLock LED to output and off (active low)
```

File: keyboards/wsk/gothic70/keymaps/default/keymap.c [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+  setPinOutput(F6);
+  writePinHigh(F6);
+  // set ScrollLock LED to output and low
```
Comment (Duckle29) [inline keyboards/wsk/gothic70/keymaps/default/keymap.c]:
```suggestion
  // set ScrollLock LED to output and off (active low)
```

File: keyboards/wsk/gothic70/keymaps/default/keymap.c [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+}
+
+void matrix_scan_user(void) {
+
+}
+
+
+uint32_t layer_state_set_user(uint32_t state)
+{
+    if (state & (1<<1)) {
```
Comment (Duckle29) [inline keyboards/wsk/gothic70/keymaps/default/keymap.c]: Consider replacing this code with a switch case statement as shown [here](https://beta.docs.qmk.fm/detailed-guides/custom_quantum_functions#layer-change-code) or at least comment the section better. Are the LEDs actually CAPS/NUMLCK/SCRLLCK, or are they layer indicator LEDs? If they're layer indicator LEDs, then the comments further up should indicate that

File: keyboards/wsk/gothic70/config.h [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+#ifdef RGB_DI_PIN
+#define RGBLIGHT_ANIMATIONS
+#define RGBLED_NUM 7
+#define RGBLIGHT_HUE_STEP 8
+#define RGBLIGHT_SAT_STEP 8
+#define RGBLIGHT_VAL_STEP 8
+#endif
+
+/* Debounce reduces chatter (unintended double-presses) - set 0 if debouncing is not needed */
+#define DEBOUNCING_DELAY 5
```
Comment (ridingqwerty) [inline keyboards/wsk/gothic70/config.h]:
```suggestion
#define DEBOUNCE 5
```

File: keyboards/wsk/gothic70/keymaps/default/keymap.c [hunk 12]
```diff
@@ -0,0 +1,83 @@
...
+  setPinOutput(F6);
+  writePinHigh(F6);
+  // set ScrollLock LED to output and low
+  setPinOutput(F7);
+  writePinHigh(F7);
+}
+
+void matrix_scan_user(void) {
+
+}
```
Comment (ridingqwerty) [inline keyboards/wsk/gothic70/keymaps/default/keymap.c]: You can remove this function since it is unused.

File: keyboards/wsk/gothic70/rules.mk [hunk 13]
```diff
@@ -0,0 +1,81 @@
...
+OPT_DEFS += -DINTERRUPT_CONTROL_ENDPOINT
+
+
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   atmega32a    bootloadHID
```
Comment (ridingqwerty) [inline keyboards/wsk/gothic70/rules.mk]: You can remove this whole section as it ought to be covered by the following `BOOTLOADER` setting.

File: keyboards/wsk/gothic70/rules.mk [hunk 14]
```diff
@@ -0,0 +1,81 @@
...
+
+
+# If you don't know the bootloader type, then you can specify the
+# Boot Section Size in *bytes* by uncommenting out the OPT_DEFS line
+#   Teensy halfKay      512
+#   Teensy++ halfKay    1024
+#   Atmel DFU loader    4096
+#   LUFA bootloader     4096
+#   USBaspLoader        2048
+# OPT_DEFS += -DBOOTLOADER_SIZE=4096
```
Comment (ridingqwerty) [inline keyboards/wsk/gothic70/rules.mk]: You can remove this.

File: keyboards/wsk/gothic70/rules.mk [hunk 15]
```diff
@@ -0,0 +1,81 @@
...
+
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration(+1000)
```
Comment (ridingqwerty) [inline keyboards/wsk/gothic70/rules.mk]:
```suggestion
BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration
```

File: keyboards/wsk/gothic70/rules.mk [hunk 16]
```diff
@@ -0,0 +1,81 @@
...
+MOUSEKEY_ENABLE = no       # Mouse keys(+4700)
```
Comment (ridingqwerty) [inline keyboards/wsk/gothic70/rules.mk]:
```suggestion
MOUSEKEY_ENABLE = no       # Mouse keys
```

File: keyboards/wsk/gothic70/rules.mk [hunk 17]
```diff
@@ -0,0 +1,81 @@
...
+EXTRAKEY_ENABLE = yes       # Audio control and System control(+450)
```
Comment (ridingqwerty) [inline keyboards/wsk/gothic70/rules.mk]:
```suggestion
EXTRAKEY_ENABLE = yes       # Audio control and System control
```

File: keyboards/wsk/gothic70/rules.mk [hunk 18]
```diff
@@ -0,0 +1,81 @@
...
+CONSOLE_ENABLE = yes        # Console for debug(+400)
```
Comment (ridingqwerty) [inline keyboards/wsk/gothic70/rules.mk]:
```suggestion
CONSOLE_ENABLE = yes        # Console for debug
```

File: keyboards/wsk/gothic70/rules.mk [hunk 19]
```diff
@@ -0,0 +1,81 @@
...
+COMMAND_ENABLE = yes        # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality on B7 by default
+RGBLIGHT_ENABLE = yes        # Enable keyboard RGB underglow
+MIDI_ENABLE = no            # MIDI support (+2400 to 4200, depending on config)
```
Comment (ridingqwerty) [inline keyboards/wsk/gothic70/rules.mk]:
```suggestion
MIDI_ENABLE = no            # MIDI support
```

File: keyboards/wsk/gothic70/rules.mk [hunk 20]
```diff
@@ -0,0 +1,81 @@
...
+UNICODE_ENABLE = no         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
+FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
+HD44780_ENABLE = no         # Enable support for HD44780 based LCDs (+400)
```
Comment (ridingqwerty) [inline keyboards/wsk/gothic70/rules.mk]:
```suggestion
HD44780_ENABLE = no         # Enable support for HD44780 based LCDs
```

File: keyboards/wsk/gothic70/keymaps/default/keymap.c [hunk 21]
```diff
@@ -0,0 +1,83 @@
...
+
+
+uint32_t layer_state_set_user(uint32_t state)
```
Comment (drashna) [inline keyboards/wsk/gothic70/keymaps/default/keymap.c]:
```suggestion
layre_state_t layer_state_set_user(layer_state_t state)
```

File: keyboards/wsk/gothic70/readme.md [hunk 22]
```diff
@@ -0,0 +1,15 @@
+# Gothic70
+
+![Gothic70](https://i.imgur.com/v84tpIQ.png)
+
+An Alice style board with arrowkeys, based on the Fate Arisu layout.
+
+Keyboard Maintainer: [Worldspawn00](https://github.com/Worldspawn00)
```
Comment (drashna) [inline keyboards/wsk/gothic70/readme.md]:
```suggestion
* Keyboard Maintainer: [Worldspawn00](https://github.com/Worldspawn00)  
```

File: keyboards/wsk/gothic70/readme.md [hunk 23]
```diff
@@ -0,0 +1,15 @@
+# Gothic70
+
+![Gothic70](https://i.imgur.com/v84tpIQ.png)
+
+An Alice style board with arrowkeys, based on the Fate Arisu layout.
+
+Keyboard Maintainer: [Worldspawn00](https://github.com/Worldspawn00)  
+Hardware Supported: Gothic70 PCB
```
Comment (drashna) [inline keyboards/wsk/gothic70/readme.md]:
```suggestion
* Hardware Supported: Gothic70 PCB
```

File: keyboards/wsk/gothic70/readme.md [hunk 24]
```diff
@@ -0,0 +1,15 @@
...
+Hardware Availability: https://geekhack.org/index.php?topic=102180.0
```
Comment (drashna) [inline keyboards/wsk/gothic70/readme.md]:
```suggestion
* Hardware Availability: https://geekhack.org/index.php?topic=102180.0
```

File: keyboards/wsk/gothic70/keymaps/default/keymap.c [hunk 25]
```diff
@@ -0,0 +1,87 @@
...
+
+#define SCAPS_LG TD(TD_SCAPS)
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+  [_BASE] = LAYOUT(
+    KC_ESC,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_GRV,  KC_BSPC,          KC_PSCR, \
+    KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,             KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,          KC_INS, \
+    KC_LCTL, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,             KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT, KC_ENT,                    KC_DEL,  \
+    SCAPS_LG, KC_Z,    KC_X,    KC_C,   KC_V,    KC_B,    MO(2),  KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT,          KC_UP,                     \
+    KC_LCTL, KC_LGUI,  KC_LALT,         KC_SPC,  MO(1),                        KC_BSPC,           KC_RALT, KC_RCTL,          KC_LEFT, KC_DOWN, KC_RGHT           \
```
Comment (Duckle29) [inline keyboards/wsk/gothic70/keymaps/default/keymap.c]:
```suggestion
    KC_LCTL, KC_LGUI,  KC_LALT,         KC_SPC,  MO(_FN),                        KC_BSPC,           KC_RALT, KC_RCTL,          KC_LEFT, KC_DOWN, KC_RGHT           \
```

File: keyboards/wsk/gothic70/keymaps/default/keymap.c [hunk 26]
```diff
@@ -0,0 +1,87 @@
...
+};
+
+#define SCAPS_LG TD(TD_SCAPS)
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+  [_BASE] = LAYOUT(
+    KC_ESC,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_GRV,  KC_BSPC,          KC_PSCR, \
+    KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,             KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,          KC_INS, \
+    KC_LCTL, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,             KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT, KC_ENT,                    KC_DEL,  \
+    SCAPS_LG, KC_Z,    KC_X,    KC_C,   KC_V,    KC_B,    MO(2),  KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT,          KC_UP,                     \
```
Comment (Duckle29) [inline keyboards/wsk/gothic70/keymaps/default/keymap.c]:
```suggestion
    SCAPS_LG, KC_Z,    KC_X,    KC_C,   KC_V,    KC_B,    MO(_MEDIA),  KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT,          KC_UP,                     \
```

**Comment by ridingqwerty** [comment]
Made some changes based on collaborator suggestions, paging @Duckle29 and @drashna for further review.

**Pull request merged**